### PR TITLE
Support configuration from bucket metadata (fixes #27)

### LINKED ISSUE #27: Support configuration on buckets
In our use case for blocklists, it would be better to be able to define the configuration on the `staging` bucket.

The groups URI could contain placeholder like this:
 `"/buckets/source/groups/{collection_id}-reviewers"`

```json
"hooks": [{
      "event": "kinto_signer.events.ReviewRequested",
      "subject": "{user_id} requested review on {bucket_id}/{collection_id}.",
      "template": "Review changes at {root_url}admin/#/buckets/{bucket_id}/collections/{collection_id}/records",
      "recipients": ["/buckets/source/groups/{collection_id}-reviewers"]
    }]
```

### PR DESCRIPTION
**Pull request by leplatrem**
Fixes #27

### CODE REVIEW

File: kinto_emailer/__init__.py [hunk 1]
```diff
@@ -48,19 +48,27 @@ def send_notification(event):
...
-def _expand_recipients(storage, recipients):
+def _get_emailer_hooks(storage, context):
+    bucket_id = context['bucket_id']
+    collection_id = context['collection_id']
+    bucket_uri = '/buckets/%s' % bucket_id
+    # Look-up collection metadata.
+    metadata = storage.get(parent_id=bucket_uri,
+                           collection_id='collection',
+                           object_id=collection_id)
+    if 'kinto-emailer' not in metadata:
```
Comment (Natim) [inline kinto_emailer/__init__.py]: 👍

File: kinto_emailer/__init__.py [hunk 2]
```diff
@@ -48,19 +48,27 @@ def send_notification(event):
...
+                               collection_id='bucket',
+                               object_id=bucket_id)
+    # Returns empty list of hooks.
+    return metadata.get('kinto-emailer', {}).get('hooks', [])
+
+
+def _expand_recipients(storage, recipients, context):
     emails = [r for r in recipients if not GROUP_REGEXP.match(r)]
-    groups = [r for r in recipients if GROUP_REGEXP.match(r)]
+    groups = [r.format(**context) for r in recipients
```
Comment (Natim) [inline kinto_emailer/__init__.py]: I think it worth having a comment saying that you might want to build the group name from the `collection_id`

File: kinto_emailer/__init__.py [hunk 3]
```diff
@@ -161,7 +170,7 @@ def includeme(config):
 
     # Listen to collection modification before commit for validation.
     config.add_subscriber(_validate_emailer_settings, ResourceChanged,
-                          for_resources=('collection',),
+                          for_resources=('bucket', 'collection',),
                           for_actions=('create', 'update'))
 
     # Listen to collection and record change events.
```
Comment (Natim) [inline kinto_emailer/__init__.py]: I guess we should probably support groups and buckets events for email notifications.

File: kinto_emailer/__init__.py (hunk 2 above)
Comment (leplatrem) [inline kinto_emailer/__init__.py]: I added it in the merge commit

File: kinto_emailer/__init__.py (hunk 3 above)
Comment (leplatrem) [inline kinto_emailer/__init__.py]: Created #31

**Pull request merged**